firebase_admin/project_management.py

```diff
@@ -0,0 +1,357 @@
+
+    @property
+    def project_id(self):
+        """The permanent, globally unique, user-assigned ID of the parent Firebase Project."""
+        return self._project_id
+
+
+class AndroidAppMetadata(AppMetadata):
+    """Android-specific information about an Android Firebase App."""
+
+    def __init__(self, name, app_id, display_name, project_id, package_name):
+        super(AndroidAppMetadata, self).__init__(name, app_id, display_name, project_id)
+        self._package_name = _check_is_nonempty_string(package_name, 'package_name')
+
+    @property
+    def package_name(self):
+        """The canonical package name of this Android App as it would appear in the Play Store."""
+        return self._package_name
+
+
+class _ProjectManagementService(object):
+    """Provides methods for interacting with the Firebase Project Management Service."""
+
+    BASE_URL = 'https://firebase.googleapis.com'
+    MAXIMUM_LIST_APPS_PAGE_SIZE = 1
+    ERROR_CODES = {
+        401: 'Request not authorized.',
+        403: 'Client does not have sufficient privileges.',
+        404: 'Failed to find the resource.',
+        409: 'The resource already exists.',
+        429: 'Request throttled out by the backend server.',
+        500: 'Internal server error.',
+        503: 'Backend servers are over capacity. Try again later.'
+    }
+
+    def __init__(self, app):
+        project_id = app.project_id
+        if not project_id:
+            raise ValueError(
+                'Project ID is required to access the Firebase Project Management Service. Either '
+                'set the projectId option, or use service account credentials. Alternatively, set '
+                'the GOOGLE_CLOUD_PROJECT environment variable.')
+        self._project_id = project_id
+        self._client = _http_client.JsonHttpClient(
+            credential=app.credential.get_credential(),
+            base_url=_ProjectManagementService.BASE_URL)
+        self._timeout = app.options.get('httpTimeout')
+
+    def get_android_app_metadata(self, app_id):
+        """Retrieves detailed information about an Android App."""
+        _check_is_nonempty_string(app_id, 'app_id')
+        path = '/v1beta1/projects/-/androidApps/{0}'.format(app_id)
+        response = self._make_request('get', path, app_id, 'App ID')
+        return AndroidAppMetadata(
+            name=response['name'],
+            app_id=response['appId'],
+            display_name=response['displayName'],
+            project_id=response['projectId'],
+            package_name=response['packageName'])
+
+    def list_android_apps(self):
+        """Lists all the Android Apps within the Firebase Project."""
+        path = '/v1beta1/projects/{0}/androidApps?pageSize={1}'.format(
+            self._project_id, _ProjectManagementService.MAXIMUM_LIST_APPS_PAGE_SIZE)
+        response = self._make_request('get', path, self._project_id, 'Project ID')
+        apps_list = []
+        while True:
```

Comment:
Is this the behavior agreed for this method during the API review? I vaguely remember something about this -- just wanted to double check. 

Otherwise we can implement this nicely with Python iterators/generators.

Reply:
Yes, for V1, we have agreed to pull the entire App list. Note that only a very small number of projects exhibit edge-case use, in which they exceed the number of allowable Apps (at most ~34 under normal use).